cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/BullseyeParser.java

```diff
@@ -200,46 +195,96 @@ private void saveConditions(CoverageMeasures fileMeasuresBuilderIn) {
     totalcoveredconditions = 0;
   }
 
-  private void updateMeasures(String kind, String event, String line, CoverageMeasures fileMeasuresBuilderIn) {
+  private static void updateMeasures(String kind, String event, String line, CoverageMeasures fileMeasuresBuilderIn) {
 
-    if ("decision".equalsIgnoreCase(kind) || "condition".equalsIgnoreCase(kind)) {
-      if ("condition".equalsIgnoreCase(kind)) {
+    switch (kind.toLowerCase(Locale.ENGLISH)) {
+      case "catch":
+      case "decision":
+      case "for-range-body":
+      case "switch-label":
+      case "try":
+        totaldecisions++;
+        setTotalCoveredDecisions(event);
+        break;
+      case "condition":
         totalconditions += 2;
-        totalcoveredconditions += 1;
+        setTotalCoveredConditions(event);
+        break;
+      case "function":
         if ("full".equalsIgnoreCase(event)) {
-          totalcoveredconditions += 1;
-        }
-        if ("none".equalsIgnoreCase(event)) {
-          totalcoveredconditions -= 1;
+          fileMeasuresBuilderIn.setHits(Integer.parseInt(line), 1);
         }
-      } else {
-        totaldecisions += 1;
+        break;
+      case "constant":
+        break;
+    default:
+        LOG.warn("BullseyeParser unknown probe kind '{}'", kind);
```

Comment:
Not sure if warning or debug only?

Reply:
This should be a warning to the user. The coverage report is not parsed accurately and SQ will report less coverage.

Reply:
The event list will not change (event has one of: none, false, true, full).

Reply:
👍